Civi/Core/ImportMap.php

```diff
@@ -0,0 +1,198 @@
+    $e->region->add([
+      'name' => 'import-map',
+      'markup' => $this->render($importMap),
+      'weight' => -1000,
+    ]);
+  }
+
+  /**
+   * Should we automatically add the import-map to the HTML header?
+   *
+   * This is currently TRUE. In the future, there may be UF services that generate the import-map.
+   *
+   * @return bool
+   */
+  public function isAutoInject(): bool {
+    return $this->autoInject;
+  }
+
+  /**
+   * Does the current page-load define any "module" files?
+   *
+   * @return bool
+   */
+  public function isRequired(): bool {
+    return $this->required;
+  }
+
+  /**
+   * @param bool $required
+   * @return ImportMap
+   */
+  public function setRequired(bool $required): ImportMap {
+    $this->required = $required;
+    return $this;
+  }
+
+  /**
+   * @param bool $autoInject
+   * @return ImportMap
+   */
+  public function setAutoInject(bool $autoInject): ImportMap {
+    $this->autoInject = $autoInject;
+    return $this;
+  }
+
+  /**
+   * Get the list of imports.
+   *
+   * @return array
+   *   Ex: ['imports' => ['square/' => 'https://example.com/square/']]
+   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap
+   * @link https://github.com/WICG/import-maps
+   */
+  public function get(): array {
+    // Just how dynamic is the import-map? Perhaps every page-view would have a different import-map?
+    // Perhaps there should be one static import-map used for all page-views? Can we cache the import-map?
+
+    // This implementation treats the import-map as a static, cacheable document (one import-map serves diverse page-views).
+    // This should prepare us for a few contingencies:
+    //   - In the future, we may encourage client-side caching (`<script type="importmap" src="https://example.com/files/cached.importmap">`).
+    //     This is defined by WICG but is not yet supported by all browsers.
+    //   - In the future, we may integrate into the import-map services of various UFs. We cannot predict whether their approach
+    //     will be static or dynamic. If our map is static, then we can safely feed it into UFs either way.
+
+    $importMap = \Civi::cache('long')->get('import-map');
+    if ($importMap === NULL) {
+      $importMap = [];
+      $importMap['imports']['civicrm/'] = \Civi::paths()->getUrl('[civicrm.root]/.');
+      \CRM_Utils_Hook::esmImportMap($importMap, []);
+      \Civi::cache('long')->set('import-map', $importMap);
+    }
+    return $importMap;
+  }
+
+  /**
+   * Format the list of imports for an HTML tag.
+   *
+   * @param array $importMap
+   *   Ex: ['imports' => ['square/' => 'https://example.com/square/']]
+   * @return string
+   *   Ex: '<script type="importmap">{"imports": ...}</script>'
+   */
+  public function render(array $importMap): string {
+    if (empty($importMap)) {
+      return '';
+    }
+    else {
+      $flags = JSON_UNESCAPED_SLASHES;
+      if (\CRM_Core_Config::singleton()->debug) {
```

Comment:
Are we trying to prefer the setting get here? I feel like that is where we are going....